sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonInputFile.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.python;
+
+import org.sonar.api.batch.fs.InputFile;
+
+public interface PythonInputFile {
+
+  InputFile wrappedFile();
+
+  boolean equals(Object obj);
+
+  int hashCode();
+
+  String toString();
```

Comment:
I don't think you need to define these at interface level, you'll still be able to override them since any object has these methods.

Reply:
What I wanted to do was enforce the implementation. Is it still bad practice to put them in the interface?

Reply:
Do we need to, though? Do we expect to create multiple `PythonInputFile` objects for the same `InputFile`?

Reply:
No, we should not. Still nothing prevents the user of the API to do that. This is why I wanted to restrict the usage a bit, but I've kept it simple in the end.